src/components/FileNameHeader.tsx

```diff
@@ -0,0 +1,136 @@
+import { Pencil1Icon } from '@radix-ui/react-icons'
+import {
+  Badge,
+  Button,
+  Dialog,
+  Flex,
+  Heading,
+  IconButton,
+  TextField,
+  Tooltip,
+} from '@radix-ui/themes'
+import { useEffect, useRef, useState } from 'react'
+import { useForm } from 'react-hook-form'
+
+import { useOverflowCheck } from '@/hooks/useOverflowCheck'
+import { useRenameFile } from '@/hooks/useRenameFile'
+import { StudioFile } from '@/types'
+import { getFileExtension } from '@/utils/file'
+
+import { FieldGroup } from './Form'
+
+interface FileNameHeaderProps {
+  file: StudioFile
+  isDirty?: boolean
+  showExt?: boolean
+}
+
+export function FileNameHeader({
```

Comment:
Not sure if it's the best name, but we only have views that represent files, so I think it's fine